joomla/joomla-cms
# Make sure the reset token is invalidated when the email is changed and an token exists

### Summary of Changes

With this change we make sure the reset token is invalidated (set to another value) when the account mail is changed.

### Testing Instructions

- Create an new user.
- request an PW reset
- check that the mail got out
- login (without resetting the PW)
- change the email (Frontend or Backend)
- try to reset the PW with the reset token send to the old mail

### Expected result

With this patch this is not possible any more as the token has been changed at the time the mail has been changed

### Actual result

Without this patch the old token still works.

### Documentation Changes Required

none.

### Other notes

This issue has been reported to the JSST by Ramesh From J2store (@rameshelamathi) but the JSST decided to fix this in the public tracker. Thanks!

cc @joomla/security @SniperSister @mbabker
